src/meta/src/manager/notification.rs

```diff
@@ -149,10 +145,14 @@ impl NotificationManager {
     }
 
     /// Tell `NotificationManagerCore` to delete sender.
-    pub async fn delete_sender(&self, worker_key: WorkerKey) {
+    pub async fn delete_sender(&self, worker_type: WorkerType, worker_key: WorkerKey) {
         let mut core_guard = self.core.lock().await;
-        core_guard.compute_senders.remove(&worker_key);
-        core_guard.frontend_senders.remove(&worker_key);
+        match worker_type {
```

Comment:
can you promise that `compute_senders` and `frontend_senders` do not intersect?
rset lgtm

Reply:
`compute_senders` are the sender who send notification to compute node, the same as the `frontend_senders`. A worker_node can't be both compute and frontend. So they should not intersect.